File: Lib/Telerik/Examples/QuickStart/Scheduler/DataBinding/CustomAppointmentEditForm.vb
Namespace Telerik.Examples.WinControls.Scheduler.DataBinding
	Partial Public Class CustomAppointmentEditForm
		Inherits Telerik.WinControls.UI.Scheduler.Dialogs.EditAppointmentDialog
		Public Sub New()
			InitializeComponent()
		End Sub

		Protected Overrides Sub LoadSettingsFromEvent(ByVal ev As Telerik.WinControls.UI.IEvent)
			MyBase.LoadSettingsFromEvent(ev)

			Dim appointmentWithEmail As AppointmentWithEmail = TryCast(ev, AppointmentWithEmail)
			If appointmentWithEmail IsNot Nothing Then
				Me.txtEmail.Text = appointmentWithEmail.Email
			End If
		End Sub

		Protected Overrides Sub ApplySettingsToEvent(ByVal ev As Telerik.WinControls.UI.IEvent)
			Dim appointmentWithEmail As AppointmentWithEmail = TryCast(ev, AppointmentWithEmail)
			If appointmentWithEmail IsNot Nothing Then
				appointmentWithEmail.Email = Me.txtEmail.Text
			End If
			MyBase.ApplySettingsToEvent(ev)
		End Sub

		Protected Overrides Function CreateNewEvent() As Telerik.WinControls.UI.IEvent
			Return New AppointmentWithEmail()
		End Function
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/Scheduler/DataBinding/CustomAppointmentEditForm.cs
namespace Telerik.Examples.WinControls.Scheduler.DataBinding
{
	public partial class CustomAppointmentEditForm : Telerik.WinControls.UI.Scheduler.Dialogs.EditAppointmentDialog
	{
		public CustomAppointmentEditForm()
		{
			InitializeComponent();
		}

		protected override void LoadSettingsFromEvent(Telerik.WinControls.UI.IEvent ev)
		{
			base.LoadSettingsFromEvent(ev);
			
			AppointmentWithEmail appointmentWithEmail = ev as AppointmentWithEmail;
			if(appointmentWithEmail != null)
			{
				this.txtEmail.Text = appointmentWithEmail.Email;
			}
		}

		protected override void ApplySettingsToEvent(Telerik.WinControls.UI.IEvent ev)
		{
			AppointmentWithEmail appointmentWithEmail = ev as AppointmentWithEmail;
			if (appointmentWithEmail != null)
			{
				appointmentWithEmail.Email = this.txtEmail.Text;
			}
			base.ApplySettingsToEvent(ev);
		}

		protected override Telerik.WinControls.UI.IEvent CreateNewEvent()
		{
			return new AppointmentWithEmail();
		}
	}
}
